File: Client/src/types/doc.ts
type Fee = Record<string, string | number | null>;
export type Status = "available" | "away" | "busy" | "unknown";

export interface Clinic {
  id: string;
  name: string;
  fee?: Fee;
  address: string;
  contact: string;
  dayTime: string[];
  whatsApp?: string;
  facilities?: string[];
  parkingAvailable?: boolean;
  location?: Record<string, number>;
  wheelchairAccessible?: boolean;
  schedule?: Record<string, { start: string; end: string; type?: string }[]>;
}

export type DoctorActions = {
  call: () => void;
  schedule: (targetElement: EventTarget & Element) => void;
  message: () => void;
  profile: () => void;
  book: () => void;
};

export interface DrCTA {
  icon: string;
  name: keyof DoctorActions;
  label: string;
  needsDD?: boolean;
  action: () => void;
  isPrimary?: boolean;
}

export interface Doc {
  id: string;
  fee: Fee;
  name: string;
  status?: Status;
  rating?: number;
  reviews?: number;
  contact?: string;
  clinics: Clinic[];
  verified: boolean;
  lastUpdated?: number;
  onlineConsult: boolean;
  bookingEnabled: boolean;
  avgConsultTime?: number;
  queuedPatients?: number;
  primarySpecialization: string;
  secondarySpecializations: string[];
  credentials: string;
  experience?: number | null;
  nextAvailable?: string | null;
  currentlyAvailable: boolean;
  waitingTime?: string;
  office?: Record<string, string>;
}

File: Server/app/models/doc.py
from uuid import uuid4
from pydantic import BaseModel, Field, field_serializer


class Clinic(BaseModel):
    name: str
    fee: int
    address: str
    contact: str
    day_time: list[str]
    whats_app: str | int | None = None
    location: dict[str, int] | None = None


class Doctor(BaseModel):
    id: str = Field(default=str(uuid4()))
    name: str
    fee: int
    status: str | None = None
    rating: int | float
    reviews: int
    contact: str | None = None
    clinics: list[Clinic]
    verified: bool = False
    last_updated: int
    online_consult: bool
    booking_enabled: bool
    avg_consult_time: int
    queued_patients: int | None = None
    primary_specialization: str
    secondary_specializations: list[str]
    credentials: str
    experience: int | None = None
    next_available: str | None = None
    currently_available: bool
    waiting_time: str | int | None = None
    office: dict[str, str]
